.github/workflows/black-command.yml

```diff
@@ -0,0 +1,44 @@
+name: black-command
+on:
+  repository_dispatch:
+    types: [black-command]
+jobs:
+  black:
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout the pull request branch
+      - uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.PAT }}
+          repository: ${{ github.event.client_payload.pull_request.head.repo.full_name }}
+          ref: ${{ github.event.client_payload.pull_request.head.ref }}
+
+      # Setup Python environment
+      - uses: actions/setup-python@v1
+
+      # Install black
+      - name: Install black
+        run: pip install black
+
+      # Execute black in check mode
+      - name: Black
+        id: black
+        run: echo ::set-output name=format::$(black --check --quiet . || echo "true")
+
+      # Execute black and commit the change to the PR branch
+      - name: Commit to the PR branch
+        if: steps.black.outputs.format == 'true'
+        run: |
+          black .
+          git config --global user.name 'actions-bot'
+          git config --global user.email '58130806+actions-bot@users.noreply.github.com'
+          git commit -am "[black-command] fixes"
+          git push
+
+      - name: Add reaction
+        uses: peter-evans/create-or-update-comment@v1
+        with:
+          token: ${{ secrets.PAT }}
```

Comment:
Whose Personal Access Token (PAT) is this, and what scope does it have? I see it's stored as a secret in the PyGMT repo (though I can't see the secret itself), and know that it's needed to let the Github Action 'bot' do the work. Just asking because I realize at https://github.com/conda-forge/gmt-feedstock/pull/100#discussion_r501595820 that we might need to use a PAT there too. Was wondering if we could have a GMT organization-wide PAT for these Github Actions that require permission to make commits.

Reply:
> Whose Personal Access Token (PAT) is this, and what scope does it have?

It's a PAT of my own account, with the `repo` scope.  

> Was wondering if we could have a GMT organization-wide secret PAT for these Github Actions that require permission to make commits.

[It seems impossible to have an organization-wide PAT](https://stackoverflow.com/questions/31159275/how-do-i-generate-the-github-oauth-token-for-organization-accounts). The only way is someone generates a PAT and saves it as an organization-wide secret.

> Alternatively, we could look into using https://github.com/tibdex/github-app-token - i.e. make an organization GMT Github App, and use that App to generate what is basically a PAT token on the fly (rather than one of our personal ones).

Yes, it looks promising.

> I'd help out with this but haven't got admin permissions in the GMT organization.

Perhaps you can ask for admin permissions if necessary.

---

BTW, this PR doesn't work. The working one is PR https://github.com/GenericMappingTools/pygmt/pull/646.

Reply:
Ok, shall we try using tibdex/github-app-token then? The step by step instructions at https://github.com/peter-evans/create-pull-request/blob/master/docs/concepts-guidelines.md#authenticating-with-github-app-generated-tokens seem fairly straightforward, and after we make the GMT-bot Github App, we can install it here on GenericMappingTools/pygmt and at GenericMappingTools/gmt-feedstock to authenticate these bots.

And yes, I'll review things on #646 later.

Reply:
It seems that the github incident is resolved, right?